libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -532,6 +569,66 @@ static MQTTStatus_t deserializeConnack( const MQTTPacketInfo_t * const pConnack,
 
 /*-----------------------------------------------------------*/
 
+static bool subscriptionPacketSize( const MQTTSubscribeInfo_t * pSubscriptionList,
+                                    size_t subscriptionCount,
+                                    size_t * pRemainingLength,
+                                    size_t * pPacketSize,
+                                    MQTTSubscriptionType_t subscriptionType )
+{
+    bool status = true;
+    size_t i = 0, packetSize = 0;
+
+    assert( pSubscriptionList != NULL );
+    assert( subscriptionCount != 0U );
+    assert( pRemainingLength != NULL );
+    assert( pPacketSize != NULL );
+
+    /* The variable header of a subscription packet consists of a 2-byte packet
+     * identifier. */
+    packetSize += sizeof( uint16_t );
+
+    /* Sum the lengths of all subscription topic filters; add 1 byte for each
+     * subscription's QoS if type is IOT_MQTT_SUBSCRIBE. */
+    for( i = 0; i < subscriptionCount; i++ )
+    {
+        /* Add the length of the topic filter. */
+        packetSize += pSubscriptionList[ i ].topicFilterLength + sizeof( uint16_t );
+
+        /* Only SUBSCRIBE packets include the QoS. */
+        if( subscriptionType == MQTT_SUBSCRIBE )
+        {
+            packetSize += 1U;
+        }
+    }
+
+    /* At this point, the "Remaining length" has been calculated. Return error
+     * if the "Remaining length" exceeds what is allowed by MQTT 3.1.1. Otherwise,
+     * set the output parameter.*/
+    if( packetSize > MQTT_MAX_REMAINING_LENGTH )
+    {
+        IotLogError( "Subscription packet length exceeds the MQTT maximum length." );
+        status = false;
+    }
+    else
+    {
+        *pRemainingLength = packetSize;
+
+        /* Calculate the full size of the subscription packet by adding the size of the
```

Comment:
adding the size of the "Remaining length" field   --> adding the number of bytes required to encode the "Remaining length" field